src/news/types.ts

```diff
@@ -0,0 +1,9 @@
+interface News{
+    readonly  url: string;
+    readonly  headline: string;
+    readonly  datetime: string;
+    readonly  source: string;
+}
+
+export type NewsItemsProps = {readonly newsItems:News[]}
+export type NewsProps = {readonly newsData:News}
```

Comment:
I might not certain about this, but in news.tsx, can't you just put News inside of the first NewsItem function as props?